### PR TITLE
meter usage by month down to the second

### PR DESCRIPTION
**Pull request by perryr16**
#### Any background context you want to provide?
Meter usage is uploaded on a per building bases. Based on ESPM meter usage data is in one-month chunks, however, it is often uploaded across a month (April 15th to May 15th). Currently, SEED can handle sorting meter data when the start and end dates are within a single month.

Monthly meter usages should be aggregated based on partial monthly values as explained by an excerpt from Portfolio Manager Energy Star, page 6, section 2: https://portfoliomanager.energystar.gov/pdf/reference/Climate%20and%20Weather.pdf
Screen Shot 2022-04-21 at 4 06 24 PM

#### What's this PR do?
Allows SEED to sort meter usage when readings span a month. The logic finds the total number of seconds within the date ranges. Then using a linear relationship determines the reading that falls within month boundaries and adds that calculated reading to the corresponding monthly total. 

#### How should this be manually tested?
Import the property AND meter data for 
[ESPM_JCC_Custom_Download_BETTERTest.xlsx](https://github.com/SEED-platform/seed/files/8557649/ESPM_JCC_Custom_Download_BETTERTest.xlsx)

On Develop: 
view property `PM Property ID: 7815736` > meters > notice date periods are contained to a single month > interval > month > successful
view property `PM Property ID: 3207429` > meters > notice date periods are not contained to a single month > interval > month > No Data

On this branch:
view property `PM Property ID: 7815736` > meters > notice date periods are contained to a single month > interval > month > successful (ensure same readings as on develop branch)
view property `PM Property ID: 3207429` > meters > notice date periods are not contained to a single month > interval > month > successful

#### What are the relevant tickets?
#2685

### CODE REVIEW

**Comment by perryr16** [comment]
A new issue for "overlapping meter data error" should be generated

File: seed/tests/test_property_views.py [hunk 1]
```diff
@@ -1941,6 +1942,7 @@ def test_property_meter_usage_can_return_monthly_meter_readings_and_column_defs_
         self.assertCountEqual(result_dict['readings'], expectation['readings'])
         self.assertCountEqual(result_dict['column_defs'], expectation['column_defs'])
 
+    @skip('Overlapping data is not valid through ESPM. This test is no longer valid')
```
Comment (Ryoken) [inline seed/tests/test_property_views.py:1945]: Nice, didn't know about @skip yet

**Review by Ryoken — APPROVED** [review]
Looks and works great!

**Comment by anchapin** [comment]
I looked into how the energy data is being split between months and found differences between what SEED is doing, what ESPM does, and what makes sense to me. I used PM ID 3207429, electricity meter reading from 1/6/18 to 2/6/18 for the example. The spreadsheet and the table inserted below show the results of my comparison. Note that I'm not sure this matters too much unless users will be comparing the monthly values between SEED and PM.

Additionally, I included my suggested approach since if you refer to the "Day Counting" sheet of the "Comparison of Splitting Data_Chapin.xlsx" spreadsheet, you'll notice that the PM approach ends up double counting the start and end dates (i.e. the total number of days for a year period is greater than 365). However, this isn't necessarily a problem since you are just trying to figure out a way of splitting the data between two months. Therefore, my recommendation would be to revise the SEED code to match the ESPM approach.

[Comparison of Splitting Data_Chapin.xlsx](https://github.com/SEED-platform/seed/files/8572790/Comparison.of.Splitting.Data_Chapin.xlsx)

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/achapin/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/achapin/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



  | SEED | Chapin | PM
-- | -- | -- | --
Total Days | 31 | 31 | 32
Jan-18 | 26 | 25.5 | 26
Feb-18 | 5 | 5.5 | 6
Energy Jan | 37,037.42 | 36,325.16 | 35,880
Energy Feb | 7,122.58 | 7,834.84 | 8,280



</body>

</html>

**Review by anchapin — CHANGES_REQUESTED** [review]
I just found the review changes section. Refer to my comments about matching the ESPM approach to splitting the data.

**Review by anchapin — APPROVED** [review]
After discussing with @perryr16, I agree that this pull request should be approved since it effectively splits the data into monthly portions. Additionally, I will open a new issue to reflect my comments on updating SEED to have the same approach as Energy Star's Portfolio Manager tool.

**Pull request merged**